src/main/java/seedu/address/ui/MainWindow.java

```diff
@@ -176,6 +176,15 @@ private CommandResult executeCommand(String commandText) throws CommandException
                 handleSwitchView(commandResult.getTargetView().trim());
             }
 
+            if (commandResult.isUndo()) {
+                logic.getElisaStateHistory().popCommand();
+                logic.setState((ElisaStateManager) logic.getElisaStateHistory().peekCommand().deepCopy());
+            } else {
+                logic.getElisaStateHistory().pushCommand(logic.getState().deepCopy());
+            }
+
```

Comment:
Is there a reason why this is here? Perhaps it would be better to put this in undoCommand.execute()? This can remove the need for checking commandResult.isUndo()